pandas/core/indexes/base.py

```diff
@@ -4938,6 +4938,25 @@ def get_indexer_for(self, target, **kwargs):
         indexer, _ = self.get_indexer_non_unique(target)
         return indexer
 
+    def _get_indexer_non_comparable(self, target, method, unique: bool = True):
+        """
+        If we have non-comparable dtypes, there will never be any matches.
+
+        For get_indexer calls with a method, this is an _inequality_ check
+        which raises TypeError.
+        """
+        if method is not None:
+            other = self._get_other_deep(target)
+            raise TypeError(f"Cannot compare dtypes {self.dtype} and {other.dtype}")
+
+        no_matches = -1 * np.ones(target.shape, dtype=np.intp)
+        if unique:
+            # This is for get_indexer
+            return no_matches
+        else:
+            # This is for get_indexer_non_unique
+            return no_matches, no_matches
```

Comment:
@jreback theres some ambiguity as to what we're supposed to be returning here.  ATM in Index.get_indexer_non_unique we are returning two ndarrays of minus-ones, but in IntervalIndex the second one is an `np.arange(len(target))` 

The other inconsistency is that ATM PeriodIndex.get_indexer defines `no_matches = -1 * np.ones(self.shape, dtype=np.intp)` whereas i expected it to use `target.shape`

Can you clear this up for me?

Reply:
looking at this more, i increasingly think the second one should be `np.arange(target.shape, dtype=np.intp)`